### PR TITLE
EPMLABSBRN-835: BE: add Description into Resource model and support it…

### PR DESCRIPTION
**Pull request by richardforgo**
[EPMLABSBRN-835](https://jira.epam.com/jira/browse/EPMLABSBRN-835) BE: add Description into Resource model and support it in appl

**Description**:
- Added Description field for Resource
- The field is now supported in reads (added to ResourceDto)
- Added new admin endpoint POST admin/resources/{id}?description=whatever-description
for updating the description for existing Resources.
- Updated unit and integration tests

### CODE REVIEW

File: api-contract/api.raml [hunk 1]
```diff
@@ -441,6 +441,17 @@ mediaType:
             responses:
                   200:
                     example: !include samples/study_histories_dto_response.json
+    /resources:
+        /{id}:
+          post:
```
Comment (goopnigoop) [inline api-contract/api.raml:446]: Hi!. Why POST is used for an update operation, but not a PUT or PATCH?

Comment (richardforgo) [reply api-contract/api.raml]: Hey @goopnigoop, good point.
I've updated this to PATCH instead in https://github.com/Brain-up/brn/pull/1830/commits/27e1364bf64238c81ba29e3efc8df8b905b62ffa

**Comment by richardforgo** [comment]
https://github.com/Brain-up/brn/pull/1830/commits/27e1364bf64238c81ba29e3efc8df8b905b62ffa
-Modified the new endpoint to be PATCH instead of POST.
-Deleted 404 case from Integration tests (should be used for happy path only afaik)
-updated all tests with given/when/then comments

File: src/main/resources/db/migration/V04092021_835.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+alter table resource add column description varchar(255);
```
Comment (richardforgo) [inline src/main/resources/db/migration/V04092021_835.sql:1]: please take a look at the versioning. I've used the date and the ticket number for this.

Comment (ElenaSpb) [reply src/main/resources/db/migration/V04092021_835.sql:1]: it should be V220210420_835
where 20210420 - current date

Comment (richardforgo) [reply src/main/resources/db/migration/V04092021_835.sql]: updated

**Review by ElenaSpb — APPROVED** [review]

**Pull request merged**